rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/aws/sns/TbSnsNodeTest.java

```diff
@@ -0,0 +1,146 @@
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.Callable;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+@ExtendWith(MockitoExtension.class)
+class TbSnsNodeTest {
+
+    private final DeviceId DEVICE_ID = new DeviceId(UUID.fromString("fccfdf2e-6a88-4a94-81dd-5cbb557019cf"));
+    private final ListeningExecutor executor = new TestDbCallbackExecutor();
+
+    private TbSnsNode node;
+    private TbSnsNodeConfiguration config;
+
+    @Mock
+    private TbContext ctxMock;
+    @Mock
+    private AmazonSNS snsClientMock;
+    @Mock
+    private PublishResult publishResultMock;
+    @Mock
+    private ResponseMetadata responseMetadataMock;
+
+    @BeforeEach
+    void setUp() {
+        node = new TbSnsNode();
+        config = new TbSnsNodeConfiguration().defaultConfiguration();
+        ReflectionTestUtils.setField(node, "snsClient", snsClientMock);
+        ReflectionTestUtils.setField(node, "config", config);
+    }
+
+    @Test
+    void verifyDefaultConfig() {
+        assertThat(config.getTopicArnPattern()).isEqualTo("arn:aws:sns:us-east-1:123456789012:MyNewTopic");
+        assertThat(config.getAccessKeyId()).isNull();
+        assertThat(config.getSecretAccessKey()).isNull();
+        assertThat(config.getRegion()).isEqualTo("us-east-1");
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    void givenTopicNamePattern_whenOnMsg_thenTellSuccess(String topicName, TbMsgMetaData metaData, String data) {
+        config.setAccessKeyId("accessKeyId");
+        config.setSecretAccessKey("secretAccessKey");
+        config.setTopicArnPattern(topicName);
+        String messageId = "msgId-1d186a16-80c7-44b3-a245-a1fc835f20c7";
+        String requestId = "reqId-bef0799b-dde9-4aa0-855b-86bbafaeaf31";
+
+        given(ctxMock.getExternalCallExecutor()).willReturn(executor);
+        given(snsClientMock.publish(any(PublishRequest.class))).willReturn(publishResultMock);
+        given(publishResultMock.getMessageId()).willReturn(messageId);
+        given(publishResultMock.getSdkResponseMetadata()).willReturn(responseMetadataMock);
+        given(responseMetadataMock.getRequestId()).willReturn(requestId);
+
+        TbMsg msg = TbMsg.newMsg(TbMsgType.POST_TELEMETRY_REQUEST, DEVICE_ID, metaData, data);
+        node.onMsg(ctxMock, msg);
+
+        PublishRequest publishRequest = new PublishRequest()
+                .withTopicArn(TbNodeUtils.processPattern(topicName, msg))
+                .withMessage(data);
+        then(snsClientMock).should().publish(publishRequest);
+        ArgumentCaptor<TbMsg> msgArgumentCaptor = ArgumentCaptor.forClass(TbMsg.class);
+        then(ctxMock).should().tellSuccess(msgArgumentCaptor.capture());
```

Comment:
will be nice to add also next assertion before checking for metadata fields:

        TbMsg outMsg = msgArgumentCaptor.getValue();
        assertThat(outMsg).usingRecursiveComparison()
                .ignoringFields("metaData", "ctx")
                .isEqualTo(msg);

Reply:
the same for the next test.